### DISCUSSION TITLE
[Question] Open-Webui, how is it releasing memory?

### DISCUSSION THREAD

**Discussion by tilllt** [open-webui/open-webui#1046] [General]
Hey there,

i have several AI Services running (Whisper-API, Auto111/StableDiffusion, TTS-Web-GUI), which all compete for the little 12GB VRAM of our GPU. Open-Webui / Ollama are the most frequently used and it happens quite regularly, that - even if no one is actively using Ollama- there is no VRAM available for the other servies.

It seems that open-webui's detection of when a session is active (keep the model loaded) or not (free the vram) is not working reliably. How is this handled at the moment?

Cheers

**Comment by Darin755** [comment]
Does the memory get freed when you restart the docker container?

**Comment by skylerparr** [comment]
I'm running into the same issue. The memory is freed when I restart the container

**Reply by skylerparr** [reply]
This seems to be isolated to ollama. Sorry for the noise.

**Reply by tilllt** [reply]
Well my assumption is that Ollama would free the memory, when OpenWebUI is asking it to... So it might be a OpenWebUI problem after all.

**Reply by TheLustriVA** [reply]
I've restarted both and the VRAM is still not being released, so I'm leaning towards webui asking ollama to keep the model loaded. Ollama drops models as soon as I restart it, usually.

**Comment by TheLustriVA** [comment]
Is there a way to free the VRAM manually? I'm going to have to uninstall open-webui at this rate.
